From 80795d56efb338bf2e4865c49928125e80dd5419 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Tue, 16 Aug 2016 16:54:36 +0800
Subject: [PATCH] applets/systemmonitor/diskactivity: regex filter

Use regex to filter out all non-physical (or "real") storage devices.

Signed-off-by: Mingcong Bai <jeffbai@aosc.xyz>
---
 applets/systemmonitor/diskactivity/contents/ui/diskactivity.qml       | 4 ++--
 applets/systemmonitor/diskactivity/contents/ui/diskactivityConfig.qml | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/applets/systemmonitor/diskactivity/contents/ui/diskactivity.qml b/applets/systemmonitor/diskactivity/contents/ui/diskactivity.qml
index 994336f..5b8d4b5 100644
--- a/applets/systemmonitor/diskactivity/contents/ui/diskactivity.qml
+++ b/applets/systemmonitor/diskactivity/contents/ui/diskactivity.qml
@@ -29,7 +29,7 @@ Applet {
     id: root
 
     onSourceAdded: {
-        var match = source.match(/^disk\/([^\/]+)\/Rate\/wblk/);
+        var match = source.match(/^disk\/((?!loop.*|nbd.*|ram.*|zram.*)[^\/]+)\/Rate\/wblk/);
         if (match) {
             var rSource = "disk/" + match[1] + "/Rate/rblk"
             root.addSource(source, match[1], rSource, match[1]);
@@ -42,4 +42,4 @@ Applet {
                         KCoreAddons.Format.formatByteSize(data2.value * 1024));
         }
     }
-}
\ No newline at end of file
+}
diff --git a/applets/systemmonitor/diskactivity/contents/ui/diskactivityConfig.qml b/applets/systemmonitor/diskactivity/contents/ui/diskactivityConfig.qml
index 68c3dd8..e47ab02 100644
--- a/applets/systemmonitor/diskactivity/contents/ui/diskactivityConfig.qml
+++ b/applets/systemmonitor/diskactivity/contents/ui/diskactivityConfig.qml
@@ -29,7 +29,7 @@ ConfigGeneral {
     id: root
 
     onSourceAdded: {
-        var match = source.match(/^disk\/([^\/]+)\/Rate\/wblk/);
+        var match = source.match(/^disk\/((?!loop.*|nbd.*|ram.*|zram.*)[^\/]+)\/Rate\/wblk/);
         if (match) {
             root.addSource(source, match[1]);
         }
-- 
2.9.3

